x-pack/plugins/fleet/server/services/fleet_server/index.ts

```diff
@@ -23,3 +27,40 @@ export async function hasFleetServers(esClient: ElasticsearchClient) {
 
   return (res.hits.total as number) > 0;
 }
+
+export async function allFleetServerVersionsAreAtLeast(
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  version: string
+): Promise<boolean> {
+  let hasMore = true;
+  const policyIds = new Set<string>();
+  let page = 1;
+  while (hasMore) {
+    const res = await packagePolicyService.list(soClient, {
+      page: page++,
+      perPage: 20,
+      kuery: 'ingest-package-policies.package.name:fleet_server',
+    });
+
+    for (const item of res.items) {
+      policyIds.add(item.policy_id);
+    }
+
+    if (res.items.length === 0) {
+      hasMore = false;
+    }
+  }
+
+  // there must be at least one fleet server for this check to pass
+  if (policyIds.size === 0) {
```

Comment:
Shouldn't we do this check based on agents, not agent policies? Or are we supposing that if there is a policy with 8.10 fleet server, that should be enough?